### PR TITLE
[CLOUDTRUST-3326] Add fields management

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

**Review by basbeu — APPROVED** [review]

File: fields/comparison.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+func (fc *fieldsComparator) CompareValueAndFunctionForUpdate(field Field, newValue *string, oldValueFunc func(Field) []string) FieldsComparator {
+	if newValue == nil {
+		// Nothing to updated
+		return fc
+	}
+	return fc.CompareValueAndFunction(field, newValue, oldValueFunc)
+}
+
+func (fc *fieldsComparator) CompareOptionalAndFunction(field Field, newValue csjson.OptionalString, oldValueFunc func(Field) []string) FieldsComparator {
```
Comment (harture) [inline fields/comparison.go:81]: CompareOptionalAndFunctionForUpdate for naming consitency ?

**Review by harture — APPROVED** [review]

File: fields/comparison.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	} else if l == 1 {
+		return fc.CompareValues(field, newValue, &oldValues[0])
+	}
+	fc.updatedFields[field] = true
+	return fc
+}
+
+func (fc *fieldsComparator) CompareValueAndFunctionForUpdate(field Field, newValue *string, oldValueFunc func(Field) []string) FieldsComparator {
+	if newValue == nil {
+		// Nothing to updated
```
Comment (harture) [inline fields/comparison.go:75]: typo

File: fields/comparison.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		return fc
+	}
+	return fc.CompareValueAndFunction(field, newValue, oldValueFunc)
+}
+
+func (fc *fieldsComparator) CompareOptionalAndFunction(field Field, newValue csjson.OptionalString, oldValueFunc func(Field) []string) FieldsComparator {
+	if !newValue.Defined {
+		// Nothing to updated
```
Comment (harture) [inline fields/comparison.go:83]: typo

**Review by harture — APPROVED** [review]

File: fields/comparison.go (hunk 1 above)
Comment (harture) [reply fields/comparison.go:81]: thks

File: fields/comparison.go (hunk 3 above)
Comment (fperot74) [reply fields/comparison.go:83]: Bastien, pfffffffff... 🤦🏻

File: fields/comparison.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		return fc
+	}
+	return fc.CompareValueAndFunction(field, newValue.Value, oldValueFunc)
+}
+
+// Return true if one of the given fields has been updated...
+// If no field is provided in parameters, just check if any field has been updated
+func (fc *fieldsComparator) IsAnyFieldUpdated(fields ...Field) bool {
```
Comment (harture) [inline fields/comparison.go:91]: I would suggest to rename to IsAnyFieldsUpdated. Maybe a method IsFieldUpdated(field Field) could also be good.

Comment (basbeu) [reply fields/comparison.go]: done

Comment (fperot74) [reply fields/comparison.go]: We are checking if ONE field in the given list has been updated => IsAnyFieldUpdated (plural would be used if it was AreAllFieldsUpdated but here, it is singular)

**Pull request merged**